train_gpt2.cu

```diff
@@ -1532,7 +1554,7 @@ int main(int argc, char *argv[]) {
     gpt2_init_common(&model);
     if (resuming == 1) {
         // if `-y 1` was set, then we are resuming from the latest checkpoint
-        gpt2_build_from_checkpoint(&model, filename_buffer);
+        gpt2_build_from_checkpoint(&model, filename_buffer, true);
```

Comment:
at this point `model.use_master_weights` is not yet initialized with `use_master_weights`, that happens below, and is just the default (true). This variable is used inside `gpt2_build_from_checkpoint` and this is probably a bug?